### PR TITLE
[ESBJAVA-5121]Test case

### PR DESCRIPTION
**Pull request by sohaniwso2**
Test case for preserving the order of the auth headers before sending the request to the backend

### CODE REVIEW

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/nhttp/transport/test/CheckAuthHeaderOrderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package org.wso2.carbon.esb.nhttp.transport.test;
```
Comment (djkevincr) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/nhttp/transport/test/CheckAuthHeaderOrderTestCase.java]: License header?

File: integration/mediation-tests/tests-transport/src/test/resources/artifacts/ESB/nhttp/transport/auth-headers.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (djkevincr) [inline integration/mediation-tests/tests-transport/src/test/resources/artifacts/ESB/nhttp/transport/auth-headers.xml]: Can we add License header here?

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/nhttp/transport/test/CheckAuthHeaderOrderTestCase.java (hunk 1 above)
Comment (sohaniwso2) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/nhttp/transport/test/CheckAuthHeaderOrderTestCase.java]: Added

File: integration/mediation-tests/tests-transport/src/test/resources/artifacts/ESB/nhttp/transport/auth-headers.xml (hunk 2 above)
Comment (sohaniwso2) [inline integration/mediation-tests/tests-transport/src/test/resources/artifacts/ESB/nhttp/transport/auth-headers.xml]: Added

**Pull request merged**